libminifi/src/SchedulingAgent.cpp

```diff
@@ -125,6 +138,19 @@ bool SchedulingAgent::onTrigger(const std::shared_ptr<core::Processor> &processo
   return false;
 }
 
+void SchedulingAgent::watchDogFunc() const {
+  std::lock_guard<std::mutex> lock(watchdog_mtx_);
+  auto now = std::chrono::steady_clock::now();
+  for (auto& info : scheduled_processors_) {
+    int64_t elapsed = std::chrono::duration_cast<std::chrono::milliseconds>(now - info.last_alert_time_).count();
+    if (elapsed > alert_time_) {
+      logger_->log_warn("%s::onTrigger has been running for %lld ms in %s", info.name_,
+          std::chrono::duration_cast<std::chrono::milliseconds>(now - info.start_time_).count(), info.uuid_);
+      info.last_alert_time_ = now;
+    }
```

Comment:
To keep the unit in the type for as long as possible, I suggest this:
```
auto elapsed = now - info.last_alert_time;
if (elapsed > alert_time_) {
  int64_t elapsed_ms{ std::chrono::duration_cast<std::chrono::milliseconds>(now - info.start_time_).count() };
  logger_->log_warn("%s::onTrigger has been running for %lld ms in %s", info.name_, elapsed_ms, info.uuid_);
  info.last_alert_time_ = now;
}
```
Using braced initializer for `elapsed` prevents narrowing implicit conversion.

Reply:
This doesn't compile:
```
/Users/aboda/work/shadow/minifi/nifi-minifi-cpp/libminifi/src/SchedulingAgent.cpp:146:17: error: invalid operands to binary expression ('std::__1::chrono::duration<long long, std::__1::ratio<1, 1000000000> >' and 'int64_t'
      (aka 'long long'))
    if (elapsed > alert_time_) {
```

Yep, crono suxx

Reply:
Done